src/world/partner/kooper.c

```diff
@@ -93,17 +151,543 @@ s32 func_802BD260_31B280(Evt* evt, s32 arg1) {
+    Camera* cam;
+    ItemEntity* itemGrabbed;
+    EncounterStatus* currentEncounter = &gCurrentEncounter;
+    PlayerStatus* playerStatus = &gPlayerStatus;
+    Npc* npc = script->owner2.npc;
+    PartnerActionStatus* partnerActionStatus = &gPartnerActionStatus;
+    CollisionStatus* collisionStatus = &gCollisionStatus;
+    f32 sp20, sp24, sp28, sp2C;
+    s32 phi_v0_4 = 0;
+    f32 temp_f20_2;
+    f32 temp_f20_3;
+    f32 temp_f20_4;
+    f32 temp_f20_5;
+    f32 temp_f20_6;
+    f32 phi_f0;
+    f32 phi_f0_3;
+    s32 tempVar;
+    f32 clamp;
+    f32 colheight;
+    
+    if (currentEncounter->unk_08 == 0) {
+        if (isInitialCall) {
+            func_802BD5F4_31B614(npc);
+            if (playerStatus->animFlags & 0x100000) {
+                return ApiStatus_DONE2;
+            }
+
+            if (D_802BEC54 == 0) {
+                tempVar = playerStatus->actionState;
+                if (tempVar < 3U) {
+                    script->functionTemp[0] = 20;
+                } else {
+                    return ApiStatus_DONE2;
+                }
+                
+            }
+
+            else if (partnerActionStatus->actionState.b[0] == 0) {
+                partnerActionStatus->actionState.b[0] = 1;
+                partnerActionStatus->actionState.b[3] = 2;
+                script->functionTemp[0] = 5;
+                npc->currentAnim.w = 0x20009;
+                D_802BEC50 = 30;
+            }
+        }
+        
+        switch (script->functionTemp[0]) {
+        case 20:
+            if (playerStatus->statusMenuCounterinputEnabledCounter == 0) {
+                if (playerStatus->decorationList == 0) {
+                    if (npc->flags & 0x1000) {
+                        disable_player_input();
+                        script->functionTemp[2] = playerStatus->statusMenuCounterinputEnabledCounter;
+                        D_802BEC64 = 1;
+                        D_802BEB40_31CB60 = 0;
+                        D_802BEC6C = 0;
+                        npc->flags &= ~0xA08;
+                        npc->flags |= 0x140;
+                        partnerActionStatus->actionState.b[3] = 2;
+                        partnerActionStatus->actionState.b[0] = 1;
+                        D_802BEC58 = func_800EF4E0();
+                        enable_npc_blur(npc);
+                        npc->duration = 4;
+                        npc->yaw = atan2(npc->pos.x, npc->pos.z, playerStatus->position.x, playerStatus->position.z);
+                        script->functionTemp[0]++;
+                    } else {
+                        return ApiStatus_DONE2;
+                    }
+                } else {
+                    return ApiStatus_DONE2;
+                }
+            } else {
+                return ApiStatus_DONE2;
+            }
+            break;
+
+        case 21:
+            if ((((u8)playerStatus->actionState - 0x15) < 3U) || (playerStatus->decorationList != 0)) {
```

Comment:
this is an or of 0x15, 0x16, 0x17. and there are enum vals for this. also same thing about decorationList from earlier